client/my-sites/domains/domain-management/components/icann-verification/icann-verification-card.jsx

```diff
@@ -2,61 +2,164 @@
+		const changeEmailHref = domainManagementEditContactInfo( selectedSiteSlug, selectedDomainName );
+
+		const { emailSent, submitting } = this.state;
+		const statusClassNames = classNames( 'icann-verification__status-container', {
+			waiting: ! emailSent,
+			sent: emailSent,
+		} );
+		let statusIcon = 'notice-outline';
+		let statusText = translate( 'Check your email — instructions sent to %(email)s.', {
+			args: { email: this.props.contactDetails.email },
+		} );
+		if ( emailSent ) {
+			statusIcon = 'mail';
+			statusText = translate( 'Sent to %(email)s. Check your email to verify.', {
+				args: { email: this.props.contactDetails.email },
+			} );
+		}
 
-	render() {
 		return (
-			<div className="icann-verification is-compact card">
-				<div className="icann-verification__explanation">
-					{ this.getExplanation() }
-				</div>
+			<div className={ statusClassNames }>
+				<div className="icann-verification__status">
+					<Gridicon icon={ statusIcon } size={ 36 } />
+					{ statusText }
 
-				<IcannVerification.Button submitting={ this.props.submitting }
-					onClick={ this.props.handleSubmit } />
+					{ ! emailSent &&
+						<div>
+							<Button
+								compact
+								busy={ submitting }
+								disabled={ submitting }
+								onClick={ this.handleSubmit }>
+								{ submitting ? translate( 'Sending…' ) : translate( 'Send Again' ) }
+							</Button>
 
-				<div className="icann-verification__explanation">
-					{ this.translate(
-						'Use this button to resend the verification email. It contains a link to verify your address. ' +
-						'{{learnMoreLink}}Learn more{{/learnMoreLink}}.', {
-							components: {
-								learnMoreLink: <a href={ support.EMAIL_VALIDATION_AND_VERIFICATION }
-									target="_blank"
-									rel="noopener noreferrer"
-								/>
-							}
-						}
-					) }
+							<Button
+								compact
+								href={ changeEmailHref }
+								onClick={ this.props.onClick }>
+								{ this.props.translate( 'Change Email Address' ) }
+							</Button>
+						</div>
+					}
 				</div>
+
 			</div>
 		);
 	}
-} );
 
-IcannVerificationCard = IcannVerification.createContainer( IcannVerificationCard );
+	render() {
+		const { selectedDomainName } = this.props;
+
+		if ( ! this.props.contactDetails ) {
+			return <QueryWhois domain={ selectedDomainName } />;
+		}
+
+		return (
+			<Card compact highlight="warning" className="icann-verification__card">
+				<QueryWhois domain={ selectedDomainName } />
+				<div className="icann-verification__explanation">
+					<h1 className="icann-verification__heading">Important: Verify Your Email Address</h1>
+					{ this.getExplanation() }
+				</div>
+
+				{ this.renderStatus() }
+			</Card>
+		);
+	}
+}
 
-export default IcannVerificationCard;
+export default connect(
+	( state, ownProps ) => ( {
+		contactDetails: getRegistrantWhois( state, ownProps.selectedDomainName ),
+	} ),
+	dispatch => bindActionCreators( { errorNotice, successNotice }, dispatch )
```

Comment:
I don't think you need to do this anymore. If you just pass
```
{ errorNotice, successNotice }
``` 
it will automagically bind: https://github.com/Automattic/wp-calypso/pull/15058/files#diff-9b3dc846fe889e4ab5d78da8b44490ddR208